modules/ollama/local.go

```diff
@@ -0,0 +1,724 @@
+			Networks: map[string]*network.EndpointSettings{},
+			NetworkSettingsBase: types.NetworkSettingsBase{
+				Bridge: "bridge",
+				Ports: nat.PortMap{
+					nat.Port(localPort + "/tcp"): {
+						{HostIP: c.host, HostPort: c.port},
+					},
+				},
+			},
+			DefaultNetworkSettings: types.DefaultNetworkSettings{
+				IPAddress: c.host,
+			},
+		},
+	}, nil
+}
+
+// IsRunning implements testcontainers.Container interface for the local Ollama binary.
+// It returns true if the local Ollama process is running, false otherwise.
+func (c *localProcess) IsRunning() bool {
+	if c.startedAt.IsZero() {
+		// The process hasn't started yet.
+		return false
+	}
+
+	select {
+	case <-c.done:
+		// The process exited.
+		return false
+	default:
+		// The process is still running.
+		return true
+	}
+}
+
+// Logs implements testcontainers.Container interface for the local Ollama binary.
+// It returns the logs from the local Ollama binary.
+func (c *localProcess) Logs(ctx context.Context) (io.ReadCloser, error) {
+	file, err := os.Open(c.logFile.Name())
+	if err != nil {
+		return nil, fmt.Errorf("open log file: %w", err)
+	}
+
+	return file, nil
+}
+
+// State implements testcontainers.Container interface for the local Ollama binary.
+// It returns the current state of the Ollama process, simulating a container state.
+func (c *localProcess) State(ctx context.Context) (*types.ContainerState, error) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
+	if !c.IsRunning() {
+		state := &types.ContainerState{
+			Status:     "exited",
+			ExitCode:   c.cmd.ProcessState.ExitCode(),
+			StartedAt:  c.startedAt.Format(time.RFC3339Nano),
+			FinishedAt: c.finishedAt.Format(time.RFC3339Nano),
+		}
+		if c.exitErr != nil {
+			state.Error = c.exitErr.Error()
+		}
+
+		return state, nil
+	}
+
+	// Setting the Running field because it's required by the wait strategy
+	// to check if the given log message is present.
+	return &types.ContainerState{
+		Status:     "running",
+		Running:    true,
+		Pid:        c.cmd.Process.Pid,
+		StartedAt:  c.startedAt.Format(time.RFC3339Nano),
+		FinishedAt: c.finishedAt.Format(time.RFC3339Nano),
+	}, nil
+}
+
+// Stop implements testcontainers.Container interface for the local Ollama binary.
+// It gracefully stops the local Ollama process.
+func (c *localProcess) Stop(ctx context.Context, d *time.Duration) error {
+	if err := c.cmd.Process.Signal(syscall.SIGTERM); err != nil {
+		return fmt.Errorf("signal ollama: %w", err)
+	}
+
+	if d != nil {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, *d)
+		defer cancel()
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-c.done:
+		// The process exited.
+		c.mtx.Lock()
+		defer c.mtx.Unlock()
+
+		return c.exitErr
+	}
+}
+
+// Terminate implements testcontainers.Container interface for the local Ollama binary.
+// It stops the local Ollama process, removing the log file.
+func (c *localProcess) Terminate(ctx context.Context) error {
```

Comment:
todo: conflict with #2926, so depending on which one is merged first, we need to update it.

Reply:
Yer just replied to that PR, I don't think it works in its current form as it breaks the interface separation.